docs/tasks/manage-stateful-set/upgrade-pet-set-to-stateful-set.md

```diff
@@ -0,0 +1,163 @@
+3. If you have the debug hook annotation `pod.alpha.kubernetes.io/initialized` set to `true`, you can remove it because it's redundant. If you don't have this annotation, you should add one, with the value set to `false`, to pause StatefulSets operations.
+
+It's recommended that you keep both PetSet manifests and StatefulSet manifests, so that you can safely roll back and recreate your PetSets, 
+if you decide not to upgrade your cluster. 
+
+#### Delete all PetSets without cascading
+
+If you find existing PetSets in your cluster in the previous step, you need to delete all PetSets *without cascading*. You can do this from `kubectl` with `--cascade=false`. 
+Note that if the flag isn't set, **cascading deletion will be performed by default**, and all Pods managed by your PetSets will be gone. 
+
+Delete those PetSets by specifying file names. This only works when
+the files contain only PetSets, but not other resources such as Services:
+
+```shell
+# Delete all existing PetSets without cascading 
+# Note that <pet-set-file> should only contain PetSets that you want to delete, but not any other resources
+kubectl delete -f <pet-set-file> --cascade=false
+```
+
+Alternatively, delete them by specifying resource names: 
+
+```shell
+# Alternatively, delete them by name and namespace without cascading
+kubectl delete petsets <pet-set-name> -n=<pet-set-namespace> --cascade=false
+```
+
+Make sure you've deleted all PetSets in the system:
+
+```shell
+# Get all PetSets again to make sure you deleted them all 
+# This should return nothing
+kubectl get petsets --all-namespaces
+```
+
+At this moment, you've deleted all PetSets in your cluster, but not their Pods, Persistent Volumes, or Persistent Volume Claims. 
+However, since the Pods are not managed by PetSets anymore, they will be vulnerable to node failures until you finish the master upgrade and recreate StatefulSets.
+
+#### Upgrade master to Kubernetes version 1.5 or later
+
+Now, you can [upgrade your Kubernetes master](/docs/admin/cluster-management/#upgrading-a-cluster) to Kubernetes version 1.5 or later.
+Note that Kubernetes nodes should NOT be upgraded at this point, since the Pods
+(that were once managed by PetSets) are now vulnerable to node failures. 
+
+#### Upgrade kubectl to Kubernetes version 1.5 or later
+
+Upgrade `kubectl` to Kubernetes version 1.5 or later, following [the steps for installing and setting up 
+kubectl](/docs/user-guide/prereqs/).
+
+#### Create StatefulSets
+
+Make sure you have both master and `kubectl` upgraded to Kubernetes version 1.5
+or later before continuing:
+
+```shell
+kubectl version
+```
+
+The output is similar to this:
+
+```shell
+Client Version: version.Info{Major:"1", Minor:"5+", GitVersion:"v1.5.0-beta.2.2+f64c9f2d999ceb-dirty", GitCommit:"f64c9f2d999ceb157d5672e9bba6639a4c456f6e", GitTreeState:"dirty", BuildDate:"2016-11-29T22:49:45Z", GoVersion:"go1.7.3", Compiler:"gc", Platform:"linux/amd64"}
```

Comment:
Here's one with a cleaner GitVersion and non-dirty tree state:

```
Client Version: version.Info{Major:"1", Minor:"5+", GitVersion:"v1.5.0-beta.2", GitCommit:"0776eab45fe28f02bbeac0f05ae1a203051a21eb", GitTreeState:"clean", BuildDate:"2016-11-24T22:35:03Z", GoVersion:"go1.7.3", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"5+", GitVersion:"v1.5.0-beta.2", GitCommit:"0776eab45fe28f02bbeac0f05ae1a203051a21eb", GitTreeState:"clean", BuildDate:"2016-11-24T22:30:23Z", GoVersion:"go1.7.3", Compiler:"gc", Platform:"linux/amd64"}
```

You could also fake it by changing `5+` to `5` and removing the `-beta.2` since these docs are supposed to target the final release. Users may be confused to see the beta stuff.